src/main/java/com/hazelcast/remotecontroller/HazelcastCloudManager.java

```diff
@@ -0,0 +1,355 @@
+        int retryCountForExceptionOfEndpoint = 0;
+        // Rarely server returns empty header, that is why a retry mechanism is added.
+        while(retryCountForExceptionOfEndpoint < 3)
+        {
+            try {
+                RequestBody body = RequestBody.create(JSON, query);
+                Request request = new Request.Builder()
+                        .url(HttpUrl.get(uri))
+                        .post(body)
+                        .header("Authorization", String.format("Bearer %s", bearerToken))
+                        .header("Content-Type", "application/json")
+                        .build();
+
+                call = client.newCall(request);
+                return call.execute();
+            }
+            catch(Exception e)
+            {
+                Log.warn(e.toString());
+            }
+            retryCountForExceptionOfEndpoint++;
+            TimeUnit.SECONDS.sleep(2);
+        }
+        throw new CloudException("Request cannot send successfully after 3 tries");
+    }
+
+    private String getTlsPassword(String clusterId) {
+        String requestUrl = String.format("%s/cluster/%s", baseUrl, clusterId);
+        try {
+            Request request = new Request.Builder()
+                    .url(requestUrl)
+                    .get()
+                    .header("Authorization", String.format("Bearer %s", bearerToken))
+                    .build();
+            call = client.newCall(request);
+            Response response = call.execute();
+            try {
+                return mapper.readTree(response.body().string()).get("tlsPassword").asText();
+            } catch (Exception e) {
+                throw new CloudException("Body is null for tlsPassword");
+            }
+        }
+        catch(Exception e)
+        {
+            LOG.warn(e.getMessage());
+            return null;
+        }
+    }
+
+    private boolean waitForStateOfCluster(String clusterId, String expectedState, long timeoutInMillisecond) throws CloudException {
+        String currentState = "";
+        long startTime = System.currentTimeMillis();
+        int retryCycle = 10;
+        try {
+            while ((System.currentTimeMillis() - startTime) < timeoutInMillisecond) {
+                String query = String.format("{ \"query\": \"query { cluster(clusterId: \\\"%s\\\") { id state } }\" }", clusterId);
+                String responseBody = prepareAndSendRequest(query).body().string();
+                LOG.info(responseBody);
+                currentState = mapper.readTree(responseBody).get("data").get("cluster").get("state").asText();
+                if(currentState.equalsIgnoreCase(expectedState))
+                    return true;
+                if(currentState.equalsIgnoreCase("FAILED"))
+                {
+                    throw new CloudException("Cluster is failed");
+                }
+                TimeUnit.SECONDS.sleep(retryCycle);
```

Comment:
Why sleep if it will be timeout?
add check like this:
```
            if ((System.currentTimeMillis() - startTime) + TimeUnit.SECONDS.toMillis(retryCycle)  > timeoutInMillisecond) {
                break;
            }
```